src/Workspaces/Core/Portable/Diagnostics/Extensions.cs

```diff
@@ -51,23 +51,28 @@ public static ImmutableArray<DiagnosticData> GetDiagnosticData(this CodeFix fix)
             return fix.Diagnostics.SelectAsArray(d => d.ToDiagnosticData(fix.Project));
         }
 
-        public static DiagnosticData ToDiagnosticData(this Diagnostic diagnostic, Project project)
+        public static DiagnosticData ToDiagnosticData(this Diagnostic diagnostic, Project project, bool isSuppressed)
```

Comment:
I think isSuppressed should be removed how about rahter than that.

create a diagnostic and then call diagnostic.WithSuppression(...) and then use the new diagnostic to existing APIs?